src/components/album/AlbumsList.js

```diff
@@ -84,7 +92,9 @@ const AlbumsList = () => {
 
   //call the function to make API request for search input value
   const handleSearch = () => {
-    fetchAlbums(searchType, searchTerm, limit);
+    setLastSearchTerm(searchTerm);
+    setCurrentPage(1);
+    fetchAlbums(searchType, searchTerm);
   };
 
   //clear search input and make an empty API call
```

Comment:
I think the below is from before your changes, but I just noticed when testing things out on this branch. We should not be setting `limit` to `1` when pressing the `Clear` button (line 104). We should also ensure that `fetchAlbums` is being called with a valid page numb er (line 106; I think we want to call `fetchAlbums('albumName', '', 1)` instead of passing it `0` which causes offset to become a negative number).

Reply:
Actually, we can remove line 106 altogether. We don't need to explicitly call `fetchAlbums`, it will already get called as a result of `setSearchTerm('')` changing the searchTerm, which will trigger the `useEffect` hook on line 65 which runs the fetchAlbum call any the search term or the search type change

Reply:
Code fixed according to them recommendations